packages/core/src/shared/handleUninstall.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { join } from 'path'
+import { getLogger } from './logger/logger'
+import { telemetry } from './telemetry'
+
+/**
+ * Checks if the extension has been uninstalled by reading the .obsolete file
+ * and comparing the number of obsolete extensions with the installed extensions.
+ *
+ * @param {vscode.ExtensionContext} context - The extension context.
+ * @param {string} obsoleteFilePath - The path to the .obsolete file.
+ * @param {string} extensionsDirPath - The path to the extensions directory.
+ * @returns {void}
+ */
+async function checkExtensionUninstall(
+    context: vscode.ExtensionContext,
+    obsoleteFilePath: string,
+    extensionsDirPath: string
+): Promise<void> {
+    try {
+        const [obsoleteFileContent, extensionsDirContent] = await Promise.all([
+            vscode.workspace.fs.readFile(vscode.Uri.file(obsoleteFilePath)),
+            vscode.workspace.fs.readDirectory(vscode.Uri.file(extensionsDirPath)),
+        ])
+
+        const extensionId = context.extension.id
```

Comment:
Let's pass only the `extension.id` (maybe typed to `VSCODE_EXTENSION_ID`) to this function since its not using anything else in `context`